onnx-ops/import_onnx_tests.py

```diff
@@ -0,0 +1,196 @@
+    #       model.mlir  (torch-mlir)
+    #       input_0.npy
+    #       output_0.npy
+    #       run_module_io_flags.txt  (flagfile with --input=input_0.npy, --expected_output=)
+
+    imported_dir_path.mkdir(parents=True, exist_ok=True)
+
+    test_data_flagfile_path = imported_dir_path / "run_module_io_flags.txt"
+    test_data_flagfile_lines = []
+
+    # Convert model.onnx up to ONNX_CONVERTER_OUTPUT_MIN_VERSION if needed.
+    # TODO(scotttodd): stamp some info e.g. importer tool / version / flags used
+    original_model_path = test_dir_path / "model.onnx"
+    converted_model_path = imported_dir_path / "model.onnx"
+    original_model = onnx.load_model(original_model_path)
+    original_version = original_model.opset_import[0].version
+    if original_version < ONNX_CONVERTER_OUTPUT_MIN_VERSION:
+        try:
+            converted_model = version_converter.convert_version(
+                original_model, ONNX_CONVERTER_OUTPUT_MIN_VERSION
+            )
+            onnx.save(converted_model, converted_model_path)
+        except:
+            # Conversion failed. Do our best with the original file.
+            # TODO(scotttodd): log a warning?
+            shutil.copy(original_model_path, converted_model_path)
+    else:
+        # No conversion needed.
+        shutil.copy(original_model_path, converted_model_path)
+
+    # Import converted model.onnx to model.mlir.
+    imported_model_path = imported_dir_path / "model.mlir"
+    exec_args = [
+        "iree-import-onnx",
+        str(converted_model_path),
+        "-o",
+        str(imported_model_path),
+    ]
+    ret = subprocess.run(exec_args, capture_output=True)
+    if ret.returncode != 0:
+        # TODO(scotttodd): log ret.stdout and ret.stderr to a file/folder?
+        print(f"  {imported_dir_path.name[5:]} import failed", file=sys.stderr)
+        return False
+
+    test_data_dirs = sorted(test_dir_path.glob("test_data_set*"))
+    if len(test_data_dirs) != 1:
+        print("WARNING: unhandled 'len(test_data_dirs) != 1'")
+        return False
+
+    # Convert from:
+    #   * input/output_*.pb
+    # to:
+    #   * input/output_*.bin (little endian binary files)
+    #   * flagfile.txt with
+    #     `--input={SHAPE}x{DTYPE}=@input_*.bin` and
+    #     `--expected_output={SHAPE}x{DTYPE}=@output_*.bin`
+    test_data_dir = test_data_dirs[0]
+    test_inputs = sorted(list(test_data_dir.glob("input_*.pb")))
+    test_outputs = sorted(list(test_data_dir.glob("output_*.pb")))
+    model = onnx.load(converted_model_path)
+    for i in range(len(test_inputs)):
+        test_input = test_inputs[i]
+        type_proto = model.graph.input[i].type
+        converted_data = convert_onnx_proto_to_numpy_array(test_input, type_proto)
+        converted_type = convert_onnx_type_proto_to_iree_type_string(type_proto)
+        # TODO(scotttodd): raise exception instead of None as flow control?
+        if converted_data is None or converted_type is None:
+            return False
```

Comment:
There is a bug fix included here: checking if the converted type is `None`. Switching using Exceptions instead of a sentinel value would be better.

Old code:
https://github.com/nod-ai/SHARK-TestSuite/blob/3b22038d687a13e3c46a8575d3e465133dfcbb7e/iree_tests/onnx/import_tests.py#L140-L145
```python
        t = convert_io_proto(test_input, model.graph.input[i].type)
        ty = get_io_proto_type(model.graph.input[i].type)
        if t is None:
            return False
```